vanilla-framework/vanilla-framework-react
# 104: Add SideNav, SideNavBanner, SideNavGroup and SideNavLink components

# Done
- Added [SideNav](https://docs.vanillaframework.io/en/patterns/navigation#sidebar-navigation) component and SideNavBanner, SideNavGroup, and SideNavLink subcomponents
- Added stories/knobs to Storybook
- Added snapshot tests
- Updated Link component with `getClassName` util

# QA
- Pull code
- Run `yarn lint` and `yarn test` to ensure there are no linting or testing errors
- Run `./run serve --watch` and navigate to http://localhost:8102/
- Verify the SideNav component in Storybook matches the [Vanilla docs](https://docs.vanillaframework.io/en/patterns/navigation#sidebar-navigation)
- Check that it operates as expected (groups open/close, on small screens the burger menu expands, etc.)

# Notes
- There are some styling bugs with current Vanilla (1.6.5) that will hopefully be fixed in the next release
  - `.sidebar__content` has `position: absolute`, so it overlaps the Storybook content a bit.
  - when inside a large column (e.g. `col-12`), the title and tagline move away from the left, so a `u-no-margin--left` utility has been added to the banner row div

# Fixes
Fixes #104
